main.tf

```
@@ -261,7 +261,7 @@ resource "aws_rds_cluster_instance" "default" {
 
 resource "aws_db_subnet_group" "default" {
   count       = local.enabled ? 1 : 0
-  name        = module.this.id
+  name        = var.subnet_group_name == "" ? module.this.id : var.subnet_group_name
```

This will allow the input to be either null or empty

```suggestion
  name        = try(length(var.subnet_group_name), 0) == 0 ? module.this.id : var.subnet_group_name
```